### PR TITLE
Get CFLAGS from c files

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
#371.

This includes #373.  Disregarding that PR, this doesn't change the generated Makefiles at all, and it looks much better.

File: release-tools/metabuild.sh [hunk 1]
```diff
@@ -52,6 +52,19 @@ add_makefile_prog() {
 	fi
 }
 
+get_cpusupport_cflags() {
+	src=$1
```
Comment (cperciva) [inline release-tools/metabuild.sh:56]: You're not using `src` here.  Also, `src` is being set to the name of the `.o` file...

File: release-tools/metabuild.sh [hunk 2]
```diff
@@ -52,6 +52,19 @@ add_makefile_prog() {
...
+
+	# Get the flags from the .c file and format them.
+	str=$(grep 'CPUSUPPORT CFLAGS:' $S | cut -f 2- -d :)
+	cflags_unstripped=$(for X in ${str} ; do		\
```
Comment (cperciva) [inline release-tools/metabuild.sh:60]: This seems unnecessarily complicated.  Why not just

```
for X in ${str}; do
    printf " \${CFLAGS_%s}" "$X";
done | sed 's/^ //'
```

and omit the `cflags_unstripped` and `cpusupport_cflags` variables entirely?

**Comment by gperciva** [comment]
Revised.

**Comment by gperciva** [comment]
oops, I just noticed that I'm inconsistent between "cpusupport" and "CPUSUPPORT" in the commit messages.

**Comment by cperciva** [comment]
Along with fixing the commit message (I would say CPUSUPPORT in both cases since we're talking about compile flags), can you add something to STYLE about the magic comments at (almost) the top of files which use special cpu features?

**Comment by gperciva** [comment]
updated

**Pull request merged**